cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/DefaultArtifactInspector.java

```diff
@@ -240,14 +240,11 @@ private ArtifactClasses.Builder inspectApplications(Id.Artifact artifactId,
         // possible for 3rd party plugin artifacts to have the main class set
         return builder;
       }
-
-      Application app = (Application) mainClass.newInstance();
```

Comment:
So, you are removing the constructor validation. Can you instead of calling of constructor at least check if (a) no-arg constructor is present (b) it's public.

Reply:
Done.